dist/tools/zep_dispatch: compilation failure

#### Description
When trying to run the border router on current `master`, compilation of `zep_dispatch` fails.

#### Steps to reproduce the issue
```
$ BUILD_IN_DOCKER=1 make -C examples/networking/gnrc/border_router BOARD=nrf52840dk clean flash term
```

#### Expected results
Should not fail

#### Actual results
after flashing, during term:
```

make -C benchmark_udp
make[2]: Nothing to be done for 'all'.
make -C dose
make[2]: Nothing to be done for 'all'.
make -C ethos
make[2]: Nothing to be done for 'all'.
make -C uhcpd
make[2]: Nothing to be done for 'all'.
make -C sliptty
make[2]: Nothing to be done for 'all'.
make -C zep_dispatch
cc -g -O3 -Wall -Wextra -I/home/mikolai/TUD/Code/RIOT-build/core/lib/include -I/home/mikolai/TUD/Code/RIOT-build/cpu/native/include -I/home/mikolai/TUD/Code/RIOT-build/drivers/include -I/home/mikolai/TUD/Code/RIOT-build/sys/include -DNDEBUG   main.c topology.c zep_parser.c /home/mikolai/TUD/Code/RIOT-build/sys/net/link_layer/ieee802154/ieee802154.c /home/mikolai/TUD/Code/RIOT-build/sys/fmt/fmt.c /home/mikolai/TUD/Code/RIOT-build/sys/net/link_layer/l2util/l2util.c /home/mikolai/TUD/Code/RIOT-build/sys/libc/string.c -o bin/zep_dispatch
In file included from /home/mikolai/TUD/Code/RIOT-build/sys/net/link_layer/ieee802154/ieee802154.c:21:
/home/mikolai/TUD/Code/RIOT-build/sys/include/string_utils.h:145:20: error: static declaration of ‘explicit_bzero’ follows non-static declaration
  145 | static inline void explicit_bzero(void *dest, size_t n_bytes)
      |                    ^~~~~~~~~~~~~~
In file included from /home/mikolai/TUD/Code/RIOT-build/sys/net/link_layer/ieee802154/ieee802154.c:18:
/usr/include/string.h:466:13: note: previous declaration of ‘explicit_bzero’ with type ‘void(void *, size_t)’ {aka ‘void(void *, long unsigned int)’}
  466 | extern void explicit_bzero (void *__s, size_t __n) __THROW __nonnull ((1))
      |             ^~~~~~~~~~~~~~
In file included from /home/mikolai/TUD/Code/RIOT-build/sys/libc/string.c:20:
/home/mikolai/TUD/Code/RIOT-build/sys/include/string_utils.h:145:20: error: static declaration of ‘explicit_bzero’ follows non-static declaration
  145 | static inline void explicit_bzero(void *dest, size_t n_bytes)
      |                    ^~~~~~~~~~~~~~
In file included from /home/mikolai/TUD/Code/RIOT-build/sys/include/string_utils.h:30:
/usr/include/string.h:466:13: note: previous declaration of ‘explicit_bzero’ with type ‘void(void *, size_t)’ {aka ‘void(void *, long unsigned int)’}
  466 | extern void explicit_bzero (void *__s, size_t __n) __THROW __nonnull ((1))
      |             ^~~~~~~~~~~~~~
make[2]: *** [Makefile:30: bin/zep_dispatch] Error 1
make[1]: *** [Makefile:8: zep_dispatch] Error 2
make: *** [Makefile:173: host-tools] Error 2
```

#### Versions
master, also 2026.07-RC1, and probably in 2026.04 (not sure how it passed testing there)

### Related issues/PR
Found while testing for https://github.com/RIOT-OS/Release-Specs/issues/337. Probably related to https://github.com/RIOT-OS/RIOT/pull/22140

### Declaration of AI-Tools / LLMs usage:

AI-Tools / LLMs that were used are:
- none

--- a/doc/doxygen/riot.doxyfile
+++ b/doc/doxygen/riot.doxyfile
@@ -2591,7 +2591,8 @@ PREDEFINED             = DOXYGEN \
                          MODULE_GNRC_SIXLOWPAN \
                          MODULE_GNRC_SIXLOWPAN_ND_ROUTER \
                          MODULE_GNRC_SIXLOWPAN_ND_BORDER_ROUTER \
-                         MODULE_GNRC_UDP
+                         MODULE_GNRC_UDP \
+                         RIOT_OS=1
 
 # If the MACRO_EXPANSION and EXPAND_ONLY_PREDEF tags are set to YES then this
 # tag can be used to specify a list of macro names that should be expanded. The

--- a/sys/include/string_utils.h
+++ b/sys/include/string_utils.h
@@ -119,14 +119,15 @@ int swprintf(string_writer_t *sw, FLASH_ATTR const char *restrict format, ...);
 #endif
 
 /* explicit_bzero is provided if:
- * - glibc is used as C lib (only with board natvie)
- * - newlib is used and __BSD_VISIBLE is set
- *      - except for ESP8266, which is using an old version of newlib without it
- * - picolibc is used and __BSD_VISIBLE is set
+ * - the system is compiled for RIOT *and* any of the following:
+ *   - glibc is used as C lib (only with board native)
+ *   - newlib is used and __BSD_VISIBLE is set
+ *     - except for ESP8266, which is using an old version of newlib without it
+ *   - picolibc is used and __BSD_VISIBLE is set
  *
  * for all other cases, we provide it here
  */
-#if !defined(CPU_NATIVE) \
+#if defined(RIOT_OS) && !defined(CPU_NATIVE) \
     && !(IS_USED(MODULE_PICOLIBC) && __BSD_VISIBLE) \
     && !(IS_USED(MODULE_NEWLIB) && __BSD_VISIBLE && !defined(CPU_ESP8266))
 

